File: Exercícios/maior10.c
#include <stdio.h>
#include <locale.h>

int main()
{

    float numeros[10];
    float maior = 0, menor = 0, media = 0;

    for (int i = 0; i < 10; i++)
    {
        printf("Insira o valor dos números: ");
        scanf("%f", &numeros[i]);
    }

    for (int i = 0; i < 10; i++)
    {
        maior = numeros[0];
        menor = numeros[0];

        if (maior < numeros[i])
        {
            maior = numeros[i];
        }
        if (menor > numeros[i])
        {
            menor = numeros[i];
        }
    }

    for (int i = 0; i < 10; i++)
    {
        media = (maior + menor) / 2;
    }

    printf("\n");

    printf("O maior número é: %.1f", maior);
    printf("\nO menor número é: %.1f", menor);
    printf("\nA média desses números é: %.1f", media);

    return 0;
}

File: Exercícios/maior10.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        float[] numeros = new float[10];
        float maior, menor, media;

        for (int i = 0; i < 10; i++) {
            System.out.print("Insira o valor dos números: ");
            numeros[i] = scanner.nextFloat();
        }

        maior = numeros[0];
        menor = numeros[0];

        for (int i = 0; i < 10; i++) {
            if (numeros[i] > maior) {
                maior = numeros[i];
            }
            if (numeros[i] < menor) {
                menor = numeros[i];
            }
        }

        media = (maior + menor) / 2;

        System.out.println("\nO maior número é: " + String.format("%.1f", maior));
        System.out.println("O menor número é: " + String.format("%.1f", menor));
        System.out.println("A média desses números é: " + String.format("%.1f", media));
    }
}